File: projet_groupe4_angular/src/app/dto/auth-response.ts
export class AuthResponse {
  constructor(
    private _token: string,
    private _nom: string,
    private _prenom: string,
    private _username: string,
    private _role : string,
  ) {}

  public get token(): string {
    return this._token;
  }

  public set token(value: string) {
    this._token = value;
  }

  public get nom(): string {
    return this._nom;
  }
  public set nom(value: string) {
    this._nom = value;
  }

  public get prenom(): string {
    return this._prenom;
  }
  public set prenom(value: string) {
    this._prenom = value;
  }

  public get username(): string {
    return this._username;
  }
  public set username(value: string) {
    this._username = value;
  }
  public get role():string{
    return this._role;
  }

  public set role(value:string){
    this._role = value;
  }

}

File: projet_groupe4/src/main/java/projet_groupe4/dto/response/AuthResponse.java
package projet_groupe4.dto.response;

import projet_groupe4.model.Personne;

public class AuthResponse {
	private String token;
	private String role;
	private String nom;
	private String prenom;
	private String username;

	public AuthResponse(String token, Personne personne) {
		if (personne == null) {
			throw new IllegalArgumentException("Personne ne peut pas être null !");
		}
		this.token = token;
		this.username = personne.getMail();
		this.nom = personne.getNom();
		this.prenom = personne.getPrenom();
		if (personne instanceof projet_groupe4.model.Client) {
			this.role = "ROLE_CLIENT";
		} else if (personne instanceof projet_groupe4.model.Employe) {
			this.role = "ROLE_EMPLOYE";
		}
	}

	public String getToken() {
		return token;
	}

	public String getRole() {
		return role;
	}

	public String getNom() {
		return nom;
	}

	public String getPrenom() {
		return prenom;
	}

	public String getUsername() {
		return username;
	}
}
